vendor/k8s.io/kubernetes/pkg/util/hash/hash.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hash
+
+import (
+	"hash"
+
+	"github.com/davecgh/go-spew/spew"
```

Comment:
My concern is the dependency because it seems [has been abandoned](https://github.com/davecgh/go-spew/issues/128).

I'm wondering if we can implement a `DeepHashObject` by ourselves with its alternatives, such as  https://github.com/kr/pretty.

Reply:
It makes sense. Let me take a look.

Reply:
By the way, the k/k is going to deprecate the `go-spew`, but we can't wait for that.  Let's try to implement it by ourselves.

Reply:
https://github.com/kubernetes/kubernetes/issues/103942